plugins/BEdita/API/src/Controller/ResourcesController.php

```diff
@@ -200,16 +201,22 @@ public function index()
     /**
      * Resource URL of a newly created entity
      *
-     * @param int|string $id Saved entity id
+     * @param \Cake\Datasource\EntityInterface $entity Resource entity
+     * @param string $primaryKey Primary key name
      * @return string Requested URL
      */
-    protected function resourceUrl($id)
+    protected function resourceUrl(EntityInterface $entity, $primaryKey)
     {
+        $prefix = 'api:resources';
+        if ($entity instanceof JsonApiSerializable) {
+            $prefix = $entity->routeNamePrefix();
+        }
+
         return Router::url(
             [
-                '_name' => $this->routeNamePrefix . ':resource',
```

Comment:
Can `$routeNamePrefix` class member (see line 74) be removed now?

Reply:
Of course! Done